tests/core/usage_statistics/test_usage_statistics_data_context.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import ast
+import logging
+from unittest import mock
+
+import pytest
+
+from great_expectations.core.usage_statistics.events import UsageStatsEvents
+from great_expectations.data_context.data_context.cloud_data_context import (
+    CloudDataContext,
+)
+from great_expectations.data_context.data_context.ephemeral_data_context import (
+    EphemeralDataContext,
+)
+from great_expectations.data_context.data_context.file_data_context import (
+    FileDataContext,
+)
+from great_expectations.data_context.util import file_relative_path
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def enable_usage_stats(monkeypatch):
+    monkeypatch.delenv("GE_USAGE_STATS")
+
+
+@pytest.fixture
+def usage_stats_decorated_methods_on_abstract_data_context() -> list[str]:
+    """
+    Uses AST parsing to collect the names of all methods that are decorated with
+    `usage_statistics_enabled_method`.
```

Comment:
The usage_statistics_enabled_method could collect method names and append them to a global list/dict etc.